### PR TITLE
chore: remove legacy trade execution utility

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Removes the legacy trade execution utility now that all swappers are using the newer pattern. Includes renaming `TradeExecution2` to `TradeExecution` as part of the cleanup.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

progresses https://github.com/shapeshift/web/issues/5024 (but doesn't close it)

## Risk

Moderate risk of broken trade execution across all swappers.

## Testing

Check all swappers still work by execution trades for each. No changes have been made to quotes, calculations etc - only the actual execution of trades has been updated so issues will likely present as total inability to execute trades for various wallets and swappers.

### Engineering

☝️

### Operations

☝️

### CODE REVIEW

**Comment by woodenfurniture** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5223** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5223" target="_blank">Graphite</a> 
    * **PR #5224** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5224" target="_blank">Graphite</a> 
      * **PR #5225** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5225" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5225?utm_source=stack-comment).

**Review by 0xApotheosis — APPROVED** [review]
Code-wise 👌 will defer to @MBMaria's testing of `wood` for functional review.

File: src/components/MultiHopTrade/hooks/useTradeExecution/useTradeExecution.ts [hunk 1]
```diff
@@ -74,21 +64,16 @@ export const useTradeExecution = ({
     return cancelPollingRef.current
   }, [])
 
-  const executeTrade = useCallback(async () => {
+  const executeTrade = useCallback(() => {
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/hooks/useTradeExecution/useTradeExecution.ts:67]: 💜

File: src/lib/swapper/tradeExecution.ts [hunk 2]
```diff
@@ -99,7 +83,196 @@ export class TradeExecution implements TradeExecutionBase {
 
       return { cancelPolling }
     } catch (e) {
+      console.log(e)
```
Comment (gomesalexandre) [inline src/lib/swapper/tradeExecution.ts:86]: Log as error?
```suggestion
      console.error(e)
```

**Review by gomesalexandre — APPROVED** [review]
Conceptual stamp code-wise pending ops testing on this guy as well 🐐

**Pull request merged**